gcloud/monitoring/timeseries.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2016 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Time series for the `Google Monitoring API (V3)`_.
+
+.. _Google Monitoring API (V3):
+    https://cloud.google.com/monitoring/api/ref_v3/rest/v3/TimeSeries
+"""
+
+# Features intentionally omitted from this first version of the client library:
+#   - Creating time series.
+#   - Natural representation of distribution values.
+
+import collections
+
+import six
+
+from gcloud.monitoring.metric import Metric
+from gcloud.monitoring.resource import Resource
+
+
+class TimeSeries(collections.namedtuple(
+        'TimeSeries', 'metric resource metric_kind value_type points')):
+    """A single time series of metric values.
+
+    This is a named tuple. See :func:`collections.namedtuple`.
+
+    :type metric: :class:`~gcloud.monitoring.metric.Metric`
+    :param metric: A metric object.
+
+    :type resource: :class:`~gcloud.monitoring.resource.Resource`
+    :param resource: A resource object.
+
+    :type metric_kind: string
+    :param metric_kind:
+        The kind of measurement: ``MetricKind.GAUGE``, ``MetricKind.DELTA``,
+        or ``MetricKind.CUMULATIVE``.
+        See :class:`~gcloud.monitoring.metric.MetricKind`.
+
+    :type value_type: string
+    :param value_type:
+        The value type of the metric: ``ValueType.BOOL``, ``ValueType.INT64``,
+        ``ValueType.DOUBLE``, ``ValueType.STRING``, ``ValueType.DISTRIBUTION``,
+        or ``ValueType.MONEY``.
+        See :class:`~gcloud.monitoring.metric.ValueType`.
+
+    :type points: list of :class:`Point`
+    :param points: A list of point objects.
+    """
+
+    @property
+    def labels(self):
+        """A single dictionary with values for all the labels.
+
+        This combines ``resource.labels`` and ``metric.labels`` and also
+        adds ``"resource_type"``.
+        """
+        # pylint: disable=attribute-defined-outside-init
```

Comment:
No, it's because `self._labels` is not initialized in `__init__`. I'd prefer not to add an `__init__` just to appease the linter. I've always felt that it is slightly wrong to have to give a variable a `None` value when it is logically undefined.